src/components/transport_manager/src/transport_adapter/threaded_socket_connection.cc

```diff
@@ -229,6 +229,8 @@ void ThreadedSocketConnection::SocketConnectionDelegate::threadMain() {
 
 void ThreadedSocketConnection::SocketConnectionDelegate::exitThreadMain() {
   SDL_AUTO_TRACE();
+  connection_->terminate_flag_ = true;
+  connection_->unexpected_disconnect_ = true;
 }
```

Comment:
@vlantonov Can you please change the order  of terminate_flag_ and unexpected_disconnect) in void ThreadedSocketConnection::Abort() to match exitThreadMain()